server/plugin/api.go

```diff
@@ -439,6 +439,13 @@ func (p *Plugin) getRecordFromServiceNow(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	record.RecordType = recordType
+	if err := record.HandleNestedFields(p.getConfiguration().ServiceNowBaseURL); err != nil {
+		p.API.LogError("Invalid body", "Error", err.Error())
+		p.handleAPIError(w, &serializer.APIErrorResponse{StatusCode: http.StatusBadRequest, Message: fmt.Sprintf("Invalid body. Error: %s", err.Error())})
```

Comment:
How can we send `BadRequest`? This is the response we got from ServiceNow. 
```suggestion
		p.API.LogError("Error in handling the nested fields", "Error", err.Error())
		p.handleAPIError(w, &serializer.APIErrorResponse{StatusCode: http.StatusInternalServerError, Message: fmt.Sprintf("Error in handling the nested fields. Error: %s", err.Error())})
```